src/sentry/static/sentry/app/stores/hookStore.tsx

```diff
@@ -73,7 +88,9 @@ const HookStore = Reflux.createStore({
     this.hooks = {};
   },
 
-  add(hookName, callback) {
+  add(hookName: ValidHookNames, callback: Callback): void {
```

Comment:
Will this type error obviate the need to check for a valid hook on like 95? Or is that runtime check still required until the entire codebase is typescript?

Reply:
@markstory Very good question. Unfortunately, the runtime check on line 95 is still necessary for any non-TS code consuming the hook store. 

The `ValidHookNames` type is only useful to ensure the right literal is passed only within TypeScript files at compile-time.

Reply:
Going to be super cool once we start converting more things to typescript though.